python-avd/pyavd/_eos_designs/structured_config/connected_endpoints/ethernet_interfaces.py

```diff
@@ -84,21 +84,26 @@ def ethernet_interfaces(self: AvdStructuredConfigConnectedEndpoints) -> list | N
     def _update_ethernet_interface_cfg(self: AvdStructuredConfigConnectedEndpoints, adapter: dict, ethernet_interface: dict, connected_endpoint: dict) -> dict:
         ethernet_interface.update(
             {
-                "type": "switched",
                 "mtu": adapter.get("mtu") if self.shared_utils.platform_settings_feature_support_per_interface_mtu else None,
                 "l2_mtu": adapter.get("l2_mtu"),
                 "l2_mru": adapter.get("l2_mru"),
-                "mode": adapter.get("mode"),
-                "vlans": adapter.get("vlans"),
-                "trunk_groups": self._get_adapter_trunk_groups(adapter, connected_endpoint),
-                "native_vlan_tag": adapter.get("native_vlan_tag"),
-                "native_vlan": adapter.get("native_vlan"),
+                "switchport": {
+                    "enabled": True,
+                    "mode": adapter.get("mode"),
+                    "trunk": {
+                        "allowed_vlan": adapter.get("vlans") if adapter.get("mode") == "trunk" else None,
+                        "groups": self._get_adapter_trunk_groups(adapter, connected_endpoint),
+                        "native_vlan_tag": adapter.get("native_vlan_tag"),
+                        "native_vlan": adapter.get("native_vlan"),
+                    },
+                    "access_vlan": adapter.get("vlans") if adapter.get("mode") in ["access", "dot1q-tunnel"] else None,
```

Comment:
https://github.com/aristanetworks/avd/pull/4454#discussion_r1756784987

Reply:
I think we need to fix a couple of things here:
- The historic default behavior if `mode` is not set, is `access`, since it relied on the EOS default to be access. We should make that more clear, so add `access` as the default in the schema, and also use it as a default value on the get().
- All the logic here is very "loose". We should be more specific to only generate good useful configs. We can match on `mode` and build the correct structured configs. Any logic here should be reflected in a user-understandable way on each key in the schema. Like "only applicable when `mode` is set to `access`" or similar.

Reply:
We have created a new issue to implement more logic here. We need to add a default to the get(). So `adapter.get("mode", "access"), since that is how the old logic worked. Please update the schema with this default.